Tile Turn-Based Party Project/Assets/Scripts/Temp/TestClass.cs

```diff
@@ -20,20 +27,78 @@ public class TestClass : Character
 
     }
 
-    public override void TakeDamage(int damage, int stat) {
-        print(curStatArr.Length);
-        curStatArr[stat] -= damage;
+    public override void TileToXY(TileBehavior tile)
+    {
+        positionx = tile.GetComponent<TileBehavior>().xPosition;
+        positiony = tile.GetComponent<TileBehavior>().yPosition;
+    }
+
+    public override void TakeDamage(int damage)
+    {
+        currentHealth -= damage;
     }
 
+
     public override void Ability() {
+        //TODO: wait on more info for now.
         throw new System.NotImplementedException();
     }
 
     public override void DisplayStats() {
         //open menu for character, display stats, etc.
+        //TODO: can hold off for now, displays atk ranges currenthp/maxhp movement
+    }
+
+    public override bool IsInRange(int targetx, int targety)
+    {
+        int dist = 0;
+        dist += Mathf.Abs(targetx - positionx) + Mathf.Abs(targety - positiony);
+        if (dist >= minrange && dist <= maxrange)
+        {
+            return true;
+        }
+        else
+        {
+            return false;
+        }
     }
 
-    public override List<int[,]> GetAttackRange() {
+    public override List<int[,]> GetAttackRange()
+    {
+        List<int[,]> retlist = new List<int[,]>();
+        foreach(GameObject tile in GameObject.Find("GameManager").GetComponent<GameManager>().mapArray)
+        {
+            int tilex = tile.GetComponent<TileBehavior>().xPosition;
+            int tiley = tile.GetComponent<TileBehavior>().yPosition;
+            if (IsInRange(tilex, tiley))
+            {
+                retlist.Add(new int[tilex, tiley]);
+            }
+        }
+        return retlist;
+    }
+    public override void ondeathhandler() //need to wait on this one to decide how summoning is implemented etc
```

Comment:
Couldn't this be called under TakeDamage? All units are summoned from PlaceCharacterOnTile in GameManager and are put in a GameObject list. One for player1 and another or player2.

Reply:
will include in future push